starport/services/chain/chain.go

```diff
@@ -111,13 +112,20 @@ func CLIHomePath(path string) Option {
 	}
 }
 
-// KeyringBackend specify the keyring backend to use for the chain command
+// KeyringBackend specifies the keyring backend to use for the chain command
 func KeyringBackend(keyringBackend chaincmd.KeyringBackend) Option {
 	return func(c *Chain) {
 		c.options.keyringBackend = keyringBackend
 	}
 }
 
+// ConfigName specifies a custom config file to use
+func ConfigName(configName string) Option {
+	return func(c *Chain) {
+		c.options.ConfigName = configName
```

Comment:
Maybe we can refactor `LocateDefault` func to `Locate(root string, []string)` where `[]string` is usually set as `conf.FileNames` but when `ConfigName` option is used it is set as `[]string{configName}`. This might simplify some parts in the code.

Reply:
`LocateDefault` has a different purpose, it has a list of file that can be considered as the default configuration then if we find nothing we can properly handle the error to say that we use a default hard coded configuration.

Custom option responds to a different logic, you always have a single file and if you don't find it you always return an error.

I think it's more maintainable to separate the two if we want to change the logic to search for the default configuration in the future